lib/api/normalize.js

```diff
@@ -0,0 +1,307 @@
+                }),
+            });
+        }
+
+        return editorState;
+    },
+
+    /**
+     * Removes all styles that use unavailable types.
+     */
+    filterInlineStyle(editorState, enabledTypes) {
+        const content = editorState.getCurrentContent();
+        const blockMap = content.getBlockMap();
+
+        const blocks = blockMap.map(block => {
+            let altered = false;
+
+            const chars = block.getCharacterList().map(char => {
+                let newChar = char;
+
+                char
+                    .getStyle()
+                    .filter(type => !enabledTypes.includes(type))
+                    .forEach(type => {
+                        altered = true;
+                        newChar = CharacterMetadata.removeStyle(newChar, type);
+                    });
+
+                return newChar;
+            });
+
+            return altered ? block.set('characterList', chars) : block;
+        });
+
+        return EditorState.set(editorState, {
+            currentContent: content.merge({
+                blockMap: blockMap.merge(blocks),
+            }),
+        });
+    },
+
+    /**
+     * Resets atomic blocks to unstyled based on which entity types are enabled,
+     * and also normalises block text to a single "space" character.
+     */
+    resetAtomicBlocks(editorState, enabledTypes) {
+        const content = editorState.getCurrentContent();
+        const blockMap = content.getBlockMap();
+        let blocks = blockMap;
+
+        const normalisedBlocks = blocks
+            .filter(
+                block =>
+                    block.getType() === BLOCK_TYPE.ATOMIC &&
+                    (block.getText() !== ' ' ||
+                        block.getInlineStyleAt(0).size !== 0),
+            )
+            .map(block => {
+                // Retain only the first character, and remove all of its styles.
+                const chars = block
+                    .getCharacterList()
+                    .slice(0, 1)
+                    .map(char => {
+                        let newChar = char;
+
+                        char.getStyle().forEach(type => {
+                            newChar = CharacterMetadata.removeStyle(
+                                newChar,
+                                type,
+                            );
+                        });
+
+                        return newChar;
+                    });
+                return block.merge({
+                    text: ' ',
+                    characterList: chars,
+                });
+            });
+
+        if (normalisedBlocks.size !== 0) {
+            blocks = blockMap.merge(normalisedBlocks);
+        }
+
+        const resetBlocks = blocks
+            .filter(block => block.getType() === BLOCK_TYPE.ATOMIC)
+            .filter(block => {
+                const entityKey = block.getEntityAt(0);
+                let shouldReset = false;
+
+                if (entityKey) {
```

Comment:
I'm not sure how there could be an atomic block without an entity in real-world content, but we never know.